Fragen/Kap.8.adoc

```diff
@@ -285,15 +285,40 @@ Geht das gut?
 
 ifdef::solution[]
 .Antwort
-Für große `n` wird der CallStack vorzeitig erschöpft, so dass kein Ergebnis berechnet wird.
+Der Rekursionsanker (Abbruchsfall) wurde unglücklich gewählt. Falls `n` ungerade ist, greift die Bedingung `n == 0` nicht und führt dazu, dass der CallStack vorzeitig erschöpft wird, so dass kein Ergebnis berechnet wird.
+Besser wäre zum Beispiel:
+[source,java]
+----
+int rek(int n) {
+  if(n <= 0) return 0; /* Falls n ungerade, ist n hier nach einigen Aufrufen -1*/
+  return 2 * rek(n - 2);
+}
+----
+
 endif::solution[]
 
 ### Frage
 Bei der Rekursion unterscheidet man stets den Abbruch- und den Wiederholungsfall. Erklären Sie das am Beispiel der rekursiven Implementierung der Fakultätsfunktion.
 
 ifdef::solution[]
 .Antwort
--- TODO --
+Die Berechnung der Fakultät kann man zum Beispiel folgendermaßen implementieren:
```

Comment:
Den Code würde ich nicht so schreiben. Aber ich habe das zum Anlaß genommen, diesen Code in die Lösung im Master einzutragen.

~~~
int fact(int n) {
  if (n <= 1) return 1; // Abbruchfall
  return n * fact(n-1); // Wiederholungsfall
}
~~~